### PR TITLE
fix: Failure loading collections on frontend results in loading loop

### LINKED ISSUE #2161: Failure loading collections on frontend results in loading loop
**To Reproduce**
1. Go to /home
2. If collections.info call fails
3. Request will be made repeatedly in a tight loop until success

### PR DESCRIPTION
**Pull request by iamsaumya**
Closes #2161

### CODE REVIEW

File: app/components/Sidebar/components/Collections.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ type Props = {
 
 function Collections({ onCreateCollection }: Props) {
   const [isFetching, setFetching] = React.useState(false);
+  const [fetchError, setFetchError] = React.useState(false);
```
Comment (tommoor) [inline app/components/Sidebar/components/Collections.js:23]:
```suggestion
  const [fetchError, setFetchError] = React.useState();
```

`undefined` is better then a boolean in this case

File: app/components/Sidebar/components/Collections.js [hunk 2]
```diff
@@ -32,17 +33,22 @@ function Collections({ onCreateCollection }: Props) {
...
-      if (!collections.isLoaded && !isFetching) {
+      if (!collections.isLoaded && !isFetching && !fetchError) {
         try {
           setFetching(true);
           await collections.fetchPage({ limit: 100 });
+        } catch (error) {
+          ui.showToast(error.message + t(", try again?"), {
+            type: "error",
+          });
+          setFetchError(true);
```
Comment (tommoor) [inline app/components/Sidebar/components/Collections.js:44]:
```suggestion
          ui.showToast(t("Collections could not be loaded, please reload the app"), {
            type: "error",
          });
          setFetchError(error);
```

The error message isn't guaranteed to make sense to the end user, probably clearer to just hard code a message here. "Try again?" doesn't make as much sense for an action that was automatic to begin with.

**Review by tommoor — CHANGES_REQUESTED** [review]
Some tweaks, but this is the right approach 👍

File: app/components/Sidebar/components/Collections.js (hunk 2 above)
Comment (iamsaumya) [reply app/components/Sidebar/components/Collections.js:44]: Yes Yes, taking note that it was automatic, tru will not definitely make sense.

**Pull request merged**